File: src/frontend/articleaifront/src/models/Recommendation.ts
import { YakeResponse } from 'src/models/AnalyseResponse'

export interface Recommendations {
  actuality: number;
  hasTags: string[];
  payload: NlpPayload[]
  topSubjects: TopSubject[]
  yakeResponse: YakeResponse[]
  classesActuality: KeywordClass[]
  keywordActuality: KeywordClass[]
}

export interface KeywordActuality{
  keyword: string
  actuality: number
}

export interface TopSubject{
  name: string
  percent: number
}

export interface KeywordClass{
  name: string
  embedding: string
  classActuality: number
}

export interface NlpPayload{
  avg: number
  isGood: number
  ngram: string
  value: number
}

File: src/main/java/com/example/ArticleAI/models/Recommendation.java
package com.example.ArticleAI.models;

import lombok.*;

import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Recommendation {
    double actuality;
    List<TopSubject> topSubjects;
    List<String> hasTags;
    List<NlpResponse> payload;
    List<KeywordClass> classesActuality;
    List<YakeResponse> yakeResponse;
    List<ClassDistance> keywordActuality;
}
